packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts

```diff
@@ -117,6 +139,50 @@ export default async (req, res) => {
   res.json({ order })
 }
 
+const updateInventoryAndReservations = async (
+  fulfillments: Fulfillment[],
+  existingFulfillments: Fulfillment[],
+  context: {
+    inventoryService: ProductVariantInventoryService
+    locationId: string
+  }
+) => {
+  const { inventoryService, locationId } = context
+  fulfillments.map(async ({ id, items }) => {
+    const existingFulfillment = existingFulfillments.find((f) => f.id === id)
+
+    if (existingFulfillment) {
+      return
+    }
+
+    await inventoryService.validateInventoryAtLocation(
+      items.map(({ item, quantity }) => ({ ...item, quantity } as LineItem)),
+      locationId
+    )
+
+    await Promise.all(
+      items.map(async ({ item, quantity }) => {
+        if (!item.variant_id) {
```

Comment:
suggestion(non blocking): create a variable with the items filtered before using it here.